### PR TITLE
Added missing InstanceNorm module.

### PR DESCRIPTION
**Pull request by joverwey**
I noticed that InstanceNorm was not implemented. I tried to follow the same pattern as BatchNorm with a few exceptions.
For BatchNorm, the current implementation always uses weights and biases and always keeps track of running stats.
In the PyTorch implementation this is optional. For InstanceNorm I do need to be able to enable/disable this so I added flags for
- affine
- track_running_stats
 
like the PyTorch implementation has.

Let me know if I missed anything.

### CODE REVIEW

**Comment by joverwey** [comment]
Clippy seems to be failing in a file I didn't touch? Any idea how I can fix this?
